app/models/codec.rb

```diff
@@ -0,0 +1,23 @@
+class Codec
+
+  CODE = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".split(//)
+
+  def self.encode(i)
+    return CODE[0] if i == 0
+    s = ''
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.